src/components/RequireAuth.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import PropTypes from "prop-types";
+
+import {
+    useAutoAuthenticate,
+    useIsAuthenticated,
+    usePerformAuth,
+    usePerformSignOut,
+} from "bento-auth-js";
+
+import { Button, Empty, Layout } from "antd";
+import { LoginOutlined } from "@ant-design/icons";
+
+import SitePageLoading from "./SitePageLoading";
+
+const styles = {
+    layout: { background: "white", padding: "48px 24px" },
+    emptyImage: { height: "auto", marginBottom: "16px" },
+};
+
+const SignInIcon = React.memo(() => (
+    <div style={{ textAlign: "center" }}>
+        <LoginOutlined style={{ fontSize: 48 }} />
+    </div>
+));
+
+const RequireAuth = ({ children }) => {
+    const { isFetching: openIdConfigFetching } = useSelector((state) => state.openIdConfiguration);
+    const { isAutoAuthenticating } = useAutoAuthenticate();
+    const isAuthenticated = useIsAuthenticated();
+    const performAuth = usePerformAuth();
+    const performSignOut = usePerformSignOut();
+
+    if (openIdConfigFetching || isAutoAuthenticating) {
+        return <SitePageLoading />;
+    }
+
+    return isAuthenticated ? children : (
+        <Layout.Content style={styles.layout}>
+            <Empty
+                image={<SignInIcon />}
+                imageStyle={styles.emptyImage}
+                description="You must sign into this node to access this page."
+            >
+                {isAuthenticated ? (
```

Comment:
Won't this always be `false`, given that we are in the falsy part of the parent `isAuthenticated` ternary?

Reply:
true, this was just straight ported from the old OwnerRoute but it's redundant for sure.